app/models/thread.php

```diff
@@ -0,0 +1,114 @@
+<?php
+class Thread extends AppModel
+{
+    const MIN_STRING_LENGTH = 1;
+    const MAX_STRING_LENGTH = 80;
+
+    public $validation = array(
+       'title' => array(
+            'length' => array(
+               'validate_between', self::MIN_STRING_LENGTH, self::MAX_STRING_LENGTH,
+            ),
+        ),
+    );
+
+    public static function getAll($offset, $limit)
+    {
+        $threads = array();
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM thread LIMIT {$offset}, {$limit}");
```

Comment:
As I have read po sa error message, seems that the SimpleDBI is not treating the value as integer, it converts the values into strings, the LIMIT is expecting integer resulting in an error po.

 Uncaught exception 'PDOException' with message 'SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''0', '6'' at line 1' in /var/www/board/vendor/SimpleDBI/SimpleDBIStatement.php:19
SimpleDBI->rows('SELECT * FROM t...', Array) #4 /var/www/board/app/controllers/thread_controller.php(11): Thread::getAll(0, 6)